lib/pbench/server/utils.py

```diff
@@ -93,21 +93,47 @@ def quarantine(dest, logger, *files):
             sys.exit(102)
 
 
-class IsoTimeHelper:
+class UtcTimeHelper:
+    """
+    A helper class to work with UTC "aware" datetime objects. A "naive" object
+    (without timezone offset) will be set to UTC timezone.
+    """
+
     def __init__(self, time: datetime.datetime):
+        """
+        Capture a datetime object and; if it's "naive", set it to be UTC; if
+        it's already "aware", adjust it to UTC.
+
+        Args:
+            time:   An aware or naive datetime object
+        """
         self.utc_time = time
-        if (
-            self.utc_time.tzinfo is None
-            or self.utc_time.tzinfo.utcoffset(self.utc_time) is None
-        ):
+        if self.utc_time.utcoffset() is None:
             self.utc_time = self.utc_time.replace(tzinfo=datetime.timezone.utc)
+        else:
+            self.utc_time = self.utc_time.astimezone(datetime.timezone.utc)
```

Comment:
If `self.utc_time` is already "aware", we're going to set it to UTC, even if it is already there....

I wonder if the code would be better if we changed line 111 to just
```
        if !self.utc_time.utcoffset():
```
so that both "naive" and UTC times are treated the same way, and only "aware", non-UTC times get converted.

Reply:
I suppose technically we don't need to do *either* to an aware UTC time, but I'm not sure I really want to add an extra case. I think that I vaguely feel slightly better about "adjusting" a UTC TZ to UTC than "replacing" the TZ... though I can't entirely justify why.

Reply:
I didn't really want to add an extra case, either, which is why I suggested collapsing the "naive" and UTC cases into the same one (and, because I'm guessing that replacing the `tzinfo` is cheaper than applying the timezone offset, so we "win" if we shunt more flows down that path).

Nevertheless, it seems like it should be the case that we shouldn't have to do anything to "aware" values until we try to render them into strings, at which point we should just stipulate which timezone they should come out in and not worry about what timezone they went in in.  But, I confess, I don't grok what the requirements are for SQLalchemy...perhaps that requires actually setting the timezone to UTC for values that we are going to feed into it.